services/private_ip_manager/src/main/java/com/futurewei/alcor/privateipmanager/controller/IpAddrController.java

```diff
@@ -238,4 +238,20 @@ public IpAddrRangeRequest getIpAddrRange(@PathVariable("range_id") String rangeI
     public List<IpAddrRangeRequest> listIpAddrRange() {
         return ipAddrService.listIpAddrRange();
     }
+
+    @PostMapping("/ips/update")
+    @ResponseBody
+    @DurationStatistics
+    public IpAddrUpdateRequest updateIpAddr(@RequestBody IpAddrUpdateRequest request) throws Exception {
+        for (IpAddrRequest oldIpAddrRequest : request.getOldIpAddrRequests()) {
+            checkRangeId(oldIpAddrRequest.getRangeId());
+            checkIpAddr(oldIpAddrRequest.getIp());
+        }
+        for (IpAddrRequest newIpAddrRequest : request.getNewIpAddrRequests()) checkIpRequest(newIpAddrRequest);
+
+        List<IpAddrRequest> ipAddrRequests = ipAddrService.updateIpAddr(request);
+        IpAddrUpdateRequest ipAddrUpdateRequest = new IpAddrUpdateRequest();
+        ipAddrUpdateRequest.setOldIpAddrRequests(ipAddrRequests);
```

Comment:
Why do we set the new ip to OldIpAddrRequests? Does it cause misunderstanding?

Reply:
Logic in the previous version